File: answers/leetcode/Implement strStr()/Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """

        if len(needle) == 0:
            return 0
        for i in range(len(haystack)):
            if haystack[i] == needle[0] and haystack[i : i + len(needle)] == needle:
                return i
        return -1

File: answers/leetcode/Implement strStr()/Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) {
            return 0;
        }
        for (int i = 0; i < haystack.length(); i ++) {
            for (int j = 0; j < needle.length() && i + j < haystack.length(); j ++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    break;
                } else if (j + 1 == needle.length()) {
                    return i;
                }
            }
        }
        return -1;
    }
}